lib/order.rb

```diff
@@ -0,0 +1,66 @@
+
+require 'csv'
+require_relative 'customer'
+
+# Class that holds information on order
+class Order
+  # Constants for order's status
+  VALID_STATUS = [:pending, :paid, :processing, :shipped, :complete]
+  
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+  
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    
+    # Check for valid order status
+    raise ArgumentError, "Status is bogus" unless VALID_STATUS.include?(@fulfillment_status)
+  end
+  
+  # Calculate order's total with tax
+  def total   
+    return 0 if @products.empty?
+
+    (@products.values.sum * 1.075).round(2)
+  end
+  
+  # Add product to order
+  def add_product(product_name, product_price)
+    raise ArgumentError, "That was already ordered" if @products.keys.include?(product_name)
+    
+    @products["#{product_name}"] = product_price
+  end
+  
+  # Load data from CSV
+  def self.all
+    file = CSV.read('data/orders.csv').map(&:to_a)
+    
+    all_orders = file.map do |new_c|      
```

Comment:
See above.  You probably meant:

```suggestion
    return file.map do |new_c|      
```

```
/Users/kaida/Developer/ada/cohort-13/grocery-store/lib/order.rb:42: warning: assigned but unused variable - all_orders
```